Lib/test/_test_multiprocessing.py

```diff
@@ -4873,6 +5103,27 @@ def test_pool(self):
         o = self.manager.Pool(processes=4)
         self.run_worker(self._test_pool, o)
 
+    @classmethod
+    def _test_queue(cls, obj):
+        assert obj.qsize() == 2
+        assert obj.full()
+        assert not obj.empty()
+        assert obj.get() == 5
+        assert not obj.empty()
+        assert obj.get() == 6
+        assert obj.empty()
+
+    def test_queue(self, qname="Queue"):
+        o = getattr(self.manager, qname)(2)
+        o.put(5)
+        o.put(6)
+        self.run_worker(self._test_queue, o)
+        assert o.empty()
+        assert not o.full()
+
+    def test_joinable_queue(self):
+        self.test_queue("JoinableQueue")
+
```

Comment:
I don't understand why you need to move these 2 methods from the previous tests. The logic appears split between the mixing class and this one for no apparent reason (but I may be missing it). If you need a different logic for certain `TestSharedMemoryManagerTypes` methods/tests you can simply override them (e.g. if you want to skip them).

Reply:
Both `Queue` and `JoinableQueue`  are implemented as distributed shared memory (a `queue.Queue` object existing in one process and an `AutoProxy` object existing in the local process).  For the same reasons mentioned in my b.p.o. comment, I believe it is much clearer to not include these on `SharedMemoryManager` by default and to instead document how they may be added on-demand with a single line of code (by calling `register()`).

There is also the potential for a POSIX shared memory message queue (not currently implemented in shared_memory.py).  This is also something worth considering for the future.

Reply:
Correct, exposing `Queue` or `JoinableQueue` here would not make them faster; they would be the same as if using them via `SyncManager`.

We could simply encourage users to employ a `SyncManager` anytime they want to use a `Queue` or `Lock` but this forces the creation of two managers with a server behind each and therefore two processes instead of only one.  I think it will be easier to explain, "create one manager to oversee work and provide cleanup at the end."

Although my use of `empty()` in this example is admittedly fragile, this example shows augmenting `SyncManager` with `Queue` is really only one line of code:
```
from multiprocessing import shared_memory
import queue
shared_memory.SharedMemoryManager.register('Queue', queue.Queue)
with shared_memory.SharedMemoryManager() as smm:
    sl = smm.ShareableList("feb")  # fast, zero-copy shared memory
    q = smm.Queue()                # slow, copy+message distributed shared memory
    for letter in sl:
        q.put(letter.upper())
    while not q.empty():
        print(q.get(timeout=1))
```

Reply:
I also think we should get rid of Queue, Lock, etc. But are those compatible with `ShareableList` at all? Can I, say, safely use `ShareableList` with `AsyncManager.Lock`?

Reply:
Agreed -- I have now gotten rid of Queue, Lock, etc. in commit 395709bbfa564c528d6d8aeba780d988bb774ede.

Yes, confirmed:  ``ShareableList`` and ``SyncManager.Lock`` can be safely mixed/used together.

I was sure you meant ``SyncManager`` when your wrote ``AsyncManager`` but I still went searching just to make sure there was not some ``AsyncManager`` hiding inside ``asyncio`` somewhere that I was not thinking of.  =)